xarray/backends/api.py

```diff
@@ -1031,11 +1036,7 @@ def open_mfdataset(
             ds.close()
         raise
 
-    def multi_file_closer():
-        for closer in closers:
-            closer()
-
-    combined.set_close(multi_file_closer)
+    combined.set_close(_multi_file_closer(closers))
```

Comment:
```suggestion
    combined.set_close(partial(_multi_file_closer, closers))
```

Don't we need this? Otherwise it'll close the files before executing `set_close`

Reply:
I suspect this is the reason the tests for Windows are failing, and that this is needed, but I don't know enough about this level of object handling to feel comfortable incorporating this yet or not.

Reply:
After reading the docs here: https://docs.python.org/3/library/functools.html#functools.partial this sounds good to me. I will commit this and trigger the tests again.